### PR TITLE
chore: Updated Setup Script for Splitting Env File

### PR DESCRIPTION
**Pull request by henit-chobisa**
## Tasks
- [x] Updated `setup.sh` file to split off frontend, backend and space environment variables.
- [x] Removed Replacement Script in `Start.sh`, `Dockerfile.space` and `Dockerfile.web`
- [x] Removed Environment Variables from `docker-compose.yml`, file based env variables are given
- [x] Added `.env.example` for apiserver as well

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -21,9 +21,49 @@ echo -e "\n\e[1;38m Instructions for generating TipTap Pro Extensions Auth Token
...
 
-read -p $'\e[1;32m Please Enter Your TipTap Pro Extensions Authentication Token: \e[0m \e[1;36m' authToken
-
+read -p $'\e[1;32m Please Enter Your TipTap Pro Extensions Authentication Token: \e[0m \e[1;36m'  authToken
 
 echo "@tiptap-pro:registry=https://registry.tiptap.dev/
 //registry.tiptap.dev/:_authToken=${authToken}" > .npmrc
 
+# Create .env files in the respective directories
+touch ./web/.env
```
Comment (sriramveeraghanta) [inline setup.sh:30]: Instead of creating a new file. use `cp` to copy everything you have in `.env.example` to `.env`

File: setup.sh [hunk 2]
```diff
@@ -21,9 +21,49 @@ echo -e "\n\e[1;38m Instructions for generating TipTap Pro Extensions Auth Token
...
+touch ./apiserver/.env
+touch ./space/.env
+
+# Initialize empty strings for each .env file
+webEnv=""
```
Comment (sriramveeraghanta) [inline setup.sh:35]: please use snake casing variables

File: space/Dockerfile.space [hunk 3]
```diff
@@ -20,19 +19,16 @@ RUN yarn install --network-timeout 500000
 
 COPY --from=builder /app/out/full/ .
 COPY turbo.json turbo.json
+COPY ./space/.env .env
```
Comment (sriramveeraghanta) [inline space/Dockerfile.space:22]: Can you explain this line?

File: space/Dockerfile.space [hunk 4]
```diff
@@ -20,19 +19,16 @@ RUN yarn install --network-timeout 500000
...
 COPY replace-env-vars.sh /usr/local/bin/
 USER root
 RUN chmod +x /usr/local/bin/replace-env-vars.sh
 
-ARG NEXT_PUBLIC_API_BASE_URL=http://localhost:8000
 ARG NEXT_PUBLIC_DEPLOY_WITH_NGINX=1
-
-ENV NEXT_PUBLIC_API_BASE_URL=$NEXT_PUBLIC_API_BASE_URL BUILT_NEXT_PUBLIC_API_BASE_URL=$NEXT_PUBLIC_API_BASE_URL NEXT_PUBLIC_DEPLOY_WITH_NGINX=$NEXT_PUBLIC_DEPLOY_WITH_NGINX
+ENV NEXT_PUBLIC_DEPLOY_WITH_NGINX=$NEXT_PUBLIC_DEPLOY_WITH_NGINX
```
Comment (sriramveeraghanta) [inline space/Dockerfile.space:28]: You don't need ARG. For anyone using Nginx and docker the value will default to 1.

**Review by sriramveeraghanta — CHANGES_REQUESTED** [review]

**Comment by sriramveeraghanta** [comment]
Please dont use Capital letters in your branch name. also you branch name should be created based on context

dev/task-name 
fix/task-name
chore/task-name
promote/task-name
enhance/task-name
ui/task-name

File: setup.sh (hunk 1 above)
Comment (henit-chobisa) [reply setup.sh]: Done ✅

File: space/Dockerfile.space (hunk 3 above)
Comment (henit-chobisa) [reply space/Dockerfile.space]: You're correct, this line doesn't make sense, as `turbo prune` imports everything and would be importing the `.env` from the project folder as well. Removed it ✅

File: space/Dockerfile.space (hunk 4 above)
Comment (henit-chobisa) [reply space/Dockerfile.space]: I made the env default to 1, not taking any argument now. ✅

File: setup.sh (hunk 2 above)
Comment (henit-chobisa) [reply setup.sh]: Done. ✅

**Comment by henit-chobisa** [comment]
> Please dont use Capital letters in your branch name. also you branch name should be created based on context
> 
> dev/task-name fix/task-name chore/task-name promote/task-name enhance/task-name ui/task-name

Will take care of this from the next time. 👍🏼

**Comment by pablohashescobar** [comment]
Closing this PR as stale.

**Pull request closed without merging**